Arrowgene.Ddon.GameServer/Handler/MailSystemMailGetAllItemHandler.cs

```diff
@@ -0,0 +1,95 @@
+using Arrowgene.Ddon.GameServer.Dump;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Logging;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using System.Linq;
+using System;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class MailSystemMailGetAllItemHandler : GameRequestPacketHandler<C2SMailSystemMailGetAllItemReq, S2CMailSystemGetAllItemRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(MailSystemMailGetAllItemHandler));
+
+        public MailSystemMailGetAllItemHandler(DdonGameServer server) : base(server)
+        {
+        }
+
+        public override S2CMailSystemGetAllItemRes Handle(GameClient client, C2SMailSystemMailGetAllItemReq request)
+        {
+            var pcap = new S2CMailSystemGetAllItemRes.Serializer().Read(pcap_data);
+
+            var result = new S2CMailSystemGetAllItemRes()
+            {
+                MessageId = request.MessageId,
+            };
+            
+            bool toItemBag = false;
+            switch (request.StorageType)
+            {
+                case 13: // ItemPost   StorageType = 13
+                    toItemBag = false;
+                    break;
+                case 19: // ItemBag    StorageType = 19
+                    toItemBag = true;
+                    break;
+                case 20: // StorageBox StorageType = 20
+                    toItemBag = false;
+                    break;
+            }
+
+            S2CItemUpdateCharacterItemNtc itemUpdateNtc = new S2CItemUpdateCharacterItemNtc()
+            {
+                UpdateType = 18 // value from pcap
```

Comment:
I believe we got an enum/constants for naming these values

Reply:
What is it called? I only see stuff like this.
```c#
        // Taken from sItemManager::moveItemsFunc (0xB9F867 in the PC Dump)
        // TODO: Cleanup
        private ushort DetermineUpdateType(byte sourceGameStorageType)
        {
            switch ( sourceGameStorageType )
            {
                case 1:
                    return 49;
                case 7:
                    return 22;
                case 19:
                    return 8;
                case 20:
                    return 9;
                default:
                    return 0;
            }
        }
```

Reply:
ItemNoticeType, introduced by yourself 😛

Reply:
I don't even remember adding it :)